ex_app/lib/all_tools/external.py

```diff
@@ -74,10 +72,51 @@ def get_public_transport_route_for_coordinates(origin_lat: str, origin_lon: str,
 		json = res.json()
 		return json
 
+	@tool
+	@safe_tool
+	def get_osm_route(profile: str, origin_lat: str, origin_lon: str, destination_lat: str, destination_lon: str,):
+		"""
+		Retrieve a route between two coordinates traveled by foot, car or bike
+		:param profile: the kind of transport used to travel the route. Available are 'routed-bike', 'routed-foot', 'routed-car'
+		:param origin_lat: Latitude of the starting point
+		:param origin_lon: Longitude of the starting point
+		:param destination_lat: Latitude of the destination
+		:param destination_lon: Longitude of the destination
+		:return: a JSON description of the route and a URL to show the route on a map
+		"""
+
+		url = f'https://routing.openstreetmap.de/{profile}/route/v1/driving/{origin_lon},{origin_lat};{destination_lon},{destination_lat}?overview=false&steps=true'
+		map_url = f' https://routing.openstreetmap.de/?loc={origin_lat}%2C{origin_lon}&loc={destination_lat}%2C{destination_lon}&srv=0'
```

Comment:
The values for srv are:
* 0 for car
* 1 for bike
* 2 for foot

Currently it only generates a car link so there's a mismatch between the text response and the preview:
![image](https://github.com/user-attachments/assets/8965e541-0a2e-46d4-8d11-06be31960cdc)